app/objects/c_agent.py

```diff
@@ -270,8 +270,7 @@ async def task(self, abilities, obfuscator, facts=(), deadman=False):
             for executor in executors:
                 ex_links = [Link.load(dict(command=self.encode_string(executor.test), paw=self.paw, ability=ability,
                                            executor=executor, deadman=deadman))]
-                valid_links = await bps.remove_links_with_unset_variables(
-                    await bps.add_test_variants(links=ex_links, agent=self, facts=facts))
+                valid_links = await bps.add_test_variants(links=ex_links, agent=self, facts=facts, trim_unset_variables=True)
```

Comment:
@djlawren - Remind me, why arent we enabling 'trim_missing_requirements' here?

Reply:
I do not know the reason that links missing requirements are not removed. I just noticed that the only function being called against the generated links in the original code is `remove_links_with_unset_variables`, so I only included the `trim_unset_variables` flag to maintain compatibility. I can try adding the other trim flag when I get a chance to see if it breaks any existing tests.

Reply:
Oh and I just remembered that in order to trim links by missing requirements you need an `operation` object. I do not think we have access to one in this method where it's called.

Reply:
Roger. Thats fine if just carrying on current functionality.